packages/pwa-kit-dev/src/ssr/server/build-dev-server.test.js

```diff
@@ -677,33 +676,82 @@ describe('DevServer rendering', () => {
 })
 
 describe('DevServer service worker', () => {
-    const TestFactory = {
-        ...NoWebpackDevServerFactory,
-        _getWebpackAsset: jest.fn()
+    let tmpDir
+
+    beforeEach(async () => {
+        tmpDir = await fse.mkdtemp(path.join(os.tmpdir(), 'pwa-kit-test-'))
```

Comment:
small nit: Should this be sync? We want the folder to exist before continuing with the test, right?

Reply:
`beforeEach` works with async operations! It will wait until the function finishes before starting tests.